src/main/java/pro/sky/JavaCoredemo2/services/ExaminerServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package pro.sky.JavaCoredemo2.services;
+
+import pro.sky.JavaCoredemo2.entity.Question;
+import pro.sky.JavaCoredemo2.exceptions.NotEnoughQuestionsException;
+import org.springframework.stereotype.Service;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Random;
+import java.util.Set;
+
+@Service
+public class ExaminerServiceImpl implements ExaminerService {
+
+    private final Random random = new Random();
+
+    private final JavaQuestionService javaQuestionService;
+    private final MathQuestionService mathQuestionService;
+
+    public ExaminerServiceImpl(JavaQuestionService javaQuestionService, MathQuestionService mathQuestionService) {
```

Comment:
```suggestion
    public ExaminerServiceImpl(QuestionService javaQuestionService, QuestionService mathQuestionService) {
```
инъекцию надо всегда делать по интерфейсу, а не реализации

Reply:
также см. внимательнее задание про Qualifier аннотацию